sns/parity_math.c

```diff
@@ -914,15 +1003,15 @@ static int isal_gen_recov_coeff_tbl(uint32_t data_count, uint32_t parity_count,
 	unit_count = data_count + parity_count;
 
 	ALLOC_ARR_INFO(decode_mat, (unit_count * data_count), "decode matrix", ret);
-	if (ret != 0)
+	if (decode_mat == NULL)
 		goto fini;
 
 	ALLOC_ARR_INFO(temp_mat, (unit_count * data_count), "temp matrix", ret);
-	if (ret != 0)
+	if (temp_mat == NULL)
 		goto fini;
 
 	ALLOC_ARR_INFO(invert_mat, (unit_count * data_count), "invert matrix", ret);
-	if (ret != 0)
+	if (invert_mat == NULL)
 		goto fini;
 
 	/* Construct temp_mat (matrix that encoded remaining frags)
```

Comment:
In the below for-loop confirm the array index used in encode_mat is correct. Since I do not have the background of this I am having a hard time understanding the code. because it seems you take the byte from alive_idx_buf buffer and multiply this value with data_count. Are you sure the encode_mat array is big enough to hold it.
Similarly the decode matrix is doing the same thing.

Reply:
In our case, data_count = 4 and parity_count = 2.
unit_count = (data_count + parity_count) = 6

The encode_matrix is of size (unit_count * data_count) which is 24 bytes for us i.e. 6 rows and 4 columns
The decode_matrix is also of same size (unit_count * data_count) i.e. 24 bytes.

In below for loop, the values present in alive_idx_buf are 0 < idx < unit_count i.e.  0 < idx < 6
Hence for 'val = (data_count * idx + j)' calculation, the range would be 0 < val < 24.

Same is true for decode_matrix, temp_matrix and inver_matrix.